### PR TITLE
Add StreamingResponse class

### PR DESCRIPTION
**Pull request by tibortakacs**
This PR introduces the new `StreamingResponse` class to enable streaming HTTP response from custom routes.
Additionally, this PR fixes the existing tests, and introduces a basic test for the type conversion.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -19,7 +19,7 @@ requires = [
 
 [tool.flit.metadata.requires-extra]
 test = [
-    "mousebender",
+    "mousebender==v2.0.0",
```
Comment (uranusjr) [inline pyproject.toml]: Why is this needed? Should we upgrade?

File: src/simpleindex/__main__.py [hunk 2]
```diff
@@ -4,33 +4,45 @@
...
 
-def _build_routes(key: str, route: routes.Route) -> typing.List[Route]:
-    async def page(request: Request):
-        response = await route.get_page(request.path_params)
-        return Response(
+def _to_http_response(
+    response: typing.Union[routes.Response, routes.StreamingResponse]
+):
+    # The order is crutial due the inheritance between *Response classes
+    if isinstance(response, routes.StreamingResponse):
```
Comment (uranusjr) [inline src/simpleindex/__main__.py]: May be a good idea to move these logic into the response classes. This if-else is manually implemented polymorphism and a good refactoring candidate.

File: src/simpleindex/routes.py [hunk 3]
```diff
@@ -7,16 +7,24 @@
...
+    content: Content = b""
     status_code: int = 200
     media_type: str = "text/plain"
     headers: typing.Optional[typing.Mapping[str, str]] = None
 
 
-Params = typing.Mapping[str, typing.Any]
+@dataclasses.dataclass()
+class StreamingResponse(Response):
+    content: ContentStream = tuple()
```
Comment (uranusjr) [inline src/simpleindex/routes.py]:
```suggestion
    content: ContentStream = ()
```

File: src/simpleindex/routes.py [hunk 4]
```diff
@@ -7,16 +7,24 @@
 
 import packaging.utils
 
+Content = typing.Union[bytes, str]
+SyncContentStream = typing.Iterable[Content]
+AsyncContentStream = typing.AsyncIterable[Content]
+ContentStream = typing.Union[AsyncContentStream, SyncContentStream]
+Params = typing.Mapping[str, typing.Any]
```
Comment (uranusjr) [inline src/simpleindex/routes.py:16]: Some of these seem unused?

File: pyproject.toml (hunk 1 above)
Comment (tibortakacs) [reply pyproject.toml]: `mousebender` API has changed quite a lot, and the used function does not exist anymore. Since this was use to parse the HTML response in tests, I thought this is better than refactoring the test more, the new API does not provide the same parsing result. Happy to dive deeper if you think it is useful.

File: src/simpleindex/__main__.py (hunk 2 above)
Comment (tibortakacs) [reply src/simpleindex/__main__.py]: It is a good idea. I am just wondering that if we put a `def to_http_response(self)` in the `*Response` class than the typing would stick to starlette. I do not think it is an issue, just want to raise it.

File: src/simpleindex/routes.py (hunk 4 above)
Comment (tibortakacs) [reply src/simpleindex/routes.py:16]: I have double-checked, all of these are used in the file somewhere as type hint.

File: src/simpleindex/__main__.py (hunk 2 above)
Comment (tibortakacs) [reply src/simpleindex/__main__.py]: I have pushed a commit with this change. Let me know what you think about it.

File: pyproject.toml (hunk 1 above)
Comment (tibortakacs) [reply pyproject.toml]: Okay, I have updated the test to use the latest mousebender.

**Review by uranusjr — APPROVED** [review]

**Comment by uranusjr** [comment]
Oops, conflicts.

**Comment by tibortakacs** [comment]
@uranusjr Thank you, I think it is ready to get it merged now.
